src/app/main/component/events/services/events.service.ts

```diff
@@ -11,9 +11,36 @@ export class EventsService implements OnDestroy {
   public currentForm: PagePreviewDTO;
   private backEnd = environment.backendLink;
   private destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
+  private isAddressFillSubject: BehaviorSubject<boolean[]> = new BehaviorSubject<boolean[]>([]);
 
   constructor(private http: HttpClient) {}
 
+  public setIsAddressFill(dates: DateEvent[], submit?: boolean, init?: boolean, check?: boolean, ind?: number): void {
+    const currentValues = this.isAddressFillSubject.getValue();
+    let newArray;
+
+    if (init) {
+      newArray = [];
+    } else if (submit) {
+      newArray = dates.map((nextValue) => !(nextValue.coordinatesDto.latitude || nextValue.onlineLink));
+    } else if (check && ind !== undefined) {
+      currentValues[ind] = currentValues[ind] === undefined ? false : !(dates[ind].coordinatesDto.latitude || dates[ind].onlineLink);
+      this.isAddressFillSubject.next(currentValues);
+      return;
+    } else if (currentValues.some((el) => el === true)) {
+      newArray = currentValues.slice(0, dates.length);
+      newArray = newArray.concat(Array(dates.length - newArray.length).fill(undefined));
+    } else {
+      newArray = dates.length && !submit ? Array(dates.length).fill(undefined) : currentValues;
+    }
+
+    this.isAddressFillSubject.next(newArray);
```

Comment:
if init === false, newArray will be undefined...

Reply:
init always true in the ngOnInit lifecycle method